airbyte-ci/connectors/pipelines/tests/test_upgrade_cdk.py

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import json
+import random
+from typing import List
+
+import anyio
+import pytest
+from connector_ops.utils import Connector, ConnectorLanguage
+from pipelines.airbyte_ci.connectors.publish import pipeline as publish_pipeline
+from pipelines.airbyte_ci.connectors.upgrade_cdk import pipeline as upgrade_cdk_pipeline
+from pipelines.models.steps import StepStatus
+from pipelines.airbyte_ci.connectors.context import ConnectorContext
+
+pytestmark = [
+    pytest.mark.anyio,
+]
+
+@pytest.fixture
+def connector_with_poetry():
+    return Connector("destination-duckdb")
```

Comment:
It'd be more stable to not rely on a single connector. I suggest to iterate on the `all_connectors` fixture and return a connector which has a `pyproject.toml` file and skip otherwise.
```suggestion
def connector_with_poetry(all_connectors):
    for connector in all_connectors:
        if (connector.code_directory / "pyproject.toml").exists():
            return connector
     pytest.skip("No connector using poetry found")
```